platform/local/dnsmasq.go

```diff
@@ -147,6 +207,85 @@ func newSegment(s byte) (*Segment, error) {
 		return nil, fmt.Errorf("LinkSetUp() failed: %v", err)
 	}
 
+	// we first create a virtual ethernet pair in current network namespace
+	// based on the current process ID
+	pid := os.Getpid()
+	pair := generateVethPair(pid)
```

Comment:
Since we don't use all bits of the PID there still can be clashes. Instead of erroring out, can we wrap this in a loop to detect a clash and retry? Another option coming to my mind is to create another global OS object like a TCP socket where the listening port is unique and has the right bit size for the values we want to have unique. With `0` as port the kernel will find a free port. E.g., using the two bytes of the TCP port means the address can be calculated as `172.(0001 and first half byte).(second half byte and third half byte).(fourth half byte and either 0000 or 0001)`.

Reply:
This may help: http://jodies.de/ipcalc?host=172.16.0.0&mask1=31&mask2=